### ISSUE TITLE
Preview fails to load when used with RequireJS

### ISSUE THREAD

**Issue by a1626** [box/box-content-preview#528]
### Environment:
- Preview version: 1.20.2, 1.20.1 and 1.19.1
- Browser name and version: Chrome V62.0.3202.94 
- Operating system and version: macOS Seirra 10.12.6

### Steps to reproduce the problem:
Code used

```javascript 
var preview = new Box.Preview();
preview.show('<file-id>', '<acess_token>', {
   container: '.preview-container',
   showDownload: true
});
```

Files were imported using cdn `https://cdn01.boxcdn.net/platform/preview/1.20.1/en-US/preview.js` and css also.

### What is the expected behavior? (Screenshots can be helpful here)
Preview for PDF file

### What went wrong? (Screenshots can be helpful here)
An icon appears with text message `PDFJS is not defined`. (Please note same code is working for an image file)

**Comment by JustinHoldstock** [comment]
Hey @a1626,

We've recently released [version 1.21.0](https://cdn01.boxcdn.net/platform/preview/1.21.0/en-US/preview.js). Can you please test with it and let us know the result?

Thanks,
Justin

**Comment by a1626** [comment]
@JustinHoldstock the issue is still there.

**Comment by a1626** [comment]
The issue is coming because mine is an `angularjs` project and thus is using `requirejs`. If i include imports for `pdf.js` and `pdf_viewer.min.js` then it works. But it is not a proper solution and we are planning to support all office files and pdf in first phase and then move on to other formats and we do not feel it's practical to test for all the formats and include all the required imports.

**Comment by jeremypress** [comment]
@a1626 we have run into issues with Preview and `requirejs` in the past. We've added a ticket in our backlog to address this, thanks for reporting!

**Comment by a1626** [comment]
@jeremypress any timeline for resolution? as this is a blocker for us.

**Comment by jeremypress** [comment]
@a1626,
It looks related to this issue https://github.com/requirejs/requirejs/issues/947. 

We load the dependencies ourselves dynamically after preview has loaded. It appears that this is not playing well with requirejs. I'm still investigating how we can combine these two ways of loading scripts.

**Comment by jeremypress** [comment]
@a1626 ,
How were you including imports for `pdf.js` and `pdf_viewer.min.js`? It looks like you're going to need to require them yourself, due to the issue listed in my previous comment.

**Comment by a1626** [comment]
@jeremypress, I was using `script` tag (CDN) for all my `box` related imports.

**Comment by tonyjin** [comment]
Hey @a1626, we've identified the issue. For some file types, (e.g. documents), we rely on third party assets (e.g. pdf.js), and since we don't want to load third party assets for file types you aren't using, they're loaded separately and managed by Preview. 

Preview itself manages its own dependencies and ordering, but since some of these third party assets use Webpack or otherwise support different module systems, when you use RequireJS in your project, it triggers the RequireJS check in some of our third party assets. For example, this block is triggered in pdf.js:

`else if(typeof define === 'function' && define.amd)
		define("pdfjs-dist/build/pdf", [], factory);`

Since we don't want our developers to have to maintain and care about all of the third party libraries that we use, we came up with a solution that temporarily disables RequireJS while Preview loads its assets, and then re-enables it after we are done loading our assets.

To enable this, you'd pass in a new option to Preview: `pauseRequireJS` (naming is hard... this isn't final yet). To ensure that you don't start loading things with RequireJS when it's been disabled by Preview, you can listen to the new `assetsloaded` event we'll emit from Preview, after which it'll be safe to use RequireJS again. During our testing, we haven't been able to reproduce a situation in which Preview's momentary disabling of RequireJS has caused any issues, even when we don't rely on the `assetsloaded` event.

Please let us know if you have any problems with this approach! Our team doesn't have a ton of experience working with RequireJS :)

**Comment by a1626** [comment]
@tonyjin even i don't have much experience with `requirejs`, but with this approach my worry is what will happen to all my resources that i'm trying to load with `requirejs`

**Comment by jeremypress** [comment]
Here's the related PR: https://github.com/box/box-content-preview/pull/561

**Comment by a1626** [comment]
@jeremypress any update on when new version with #561 can be expected to be released?

**Comment by jeremypress** [comment]
@a1626 I'm hoping for next Tuesday (the 9th), but the latest should be the Tuesday after that (the 16th). I'm going to close your issue about documentation as it will be included in #561.

**Comment by jnimety** [comment]
I'm running into the same issue but we use `webpack`, not `requirejs`. Could `webpack` be affected too?

**Comment by bell87** [comment]
Also encountering this issue, loading box js libraries via cdn as per [example](https://github.com/box/box-content-preview#including-preview-as-a-library). Any suggestions / update?

**Comment by a1626** [comment]
@bell87 as per @jeremypress we should have new version release soon, but if you want immediate solution you can either

- Take his branch and build library from it (which is what i used to do one round of testing).

or

- Use the solution that i used for pdf, i.e. to include extra imports. You can read about it [here](https://community.box.com/t5/Box-Developer-Forum/PDFJS-is-not-defined/m-p/48022#M3611)

**Comment by jeremypress** [comment]
Hey @jnimety, can you open a separate issue with your webpack config? Our fix in flight will not affect webpack.

**Comment by jeremypress** [comment]
#561 is merged and will go live with our next release (targeted for Tuesday).

Thank you for your patience!

**Comment by jnimety** [comment]
@jeremypress I switched to using the CDN and everything works. Turns out the file type plugins won't load unless you use the cdn.

**Comment by pramodsum** [comment]
Fixed in [Preview v1.26.0](https://github.com/box/box-content-preview/releases/tag/v1.26.0)
